homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -284,16 +279,11 @@ async def _async_from_discovery(
                 break
             self._abort_if_unique_id_configured()
 
-        fqdn_with_ssl_verification = (
-            existing_entry
-            and not _is_valid_ip(existing_entry.data[CONF_HOST])
-            and existing_entry.data[CONF_VERIFY_SSL]
-        )
-
         if (
             existing_entry
+            and is_ip(existing_entry.data[CONF_HOST])
             and existing_entry.data[CONF_HOST] != host
-            and not fqdn_with_ssl_verification
+            and is_ipv4(existing_entry.data[CONF_HOST]) == is_ipv4(host)
```

Comment:
```suggestion
            and is_ipv4(existing_entry.data[CONF_HOST])
            and is_ipv4(host)
```

I don't think this was intended?

Reply:
it is intended - it should check if both current and discovered IP are from same version (_see changes in tests_) :)

Reply:
If you don't care if they are ipv4 and just the same version
```
        if (
            existing_entry
            and existing_entry.data[CONF_HOST] != host
            and is_ip(existing_entry.data[CONF_HOST])
            and is_ip(host)
            and ip_address(existing_entry.data[CONF_HOST]).version == ip_address(host).version
        ):
```

Reply:
as always, many thanks for your review and inputs 👍